kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -65,11 +69,19 @@ def build(self):
 
         # check T or set default
         if self.T is None:
-            self.T = 2**(self.J)
+            self._average = True
+            self.T = 2 ** self.J
         elif self.T > N_input:
             raise ValueError("The temporal support T of the low-pass filter "
                              "cannot exceed input length (got {} > {})".format(
                                  self.T, N_input))
+        elif self.T < 0:
+            raise ValueError("T must be a nonnegative integer (got {})".format(
```

Comment:
Is float T even implemented?

Reply:
yes, (although untested). `T` is used in two places in `scattering_filter_factory`

```
sigma_low = sigma0 / T
```
(float/int or float/float, assigns to float)
so `sigma_low` is float


```
log2_T = math.floor(math.log2(T))
```
(log2(int or float) assigns to float, math.floor of float assigns to int)
so `log2_T` is int

no lizard.

Reply:
> "average=False will override any T" is still not right. Here's what i have in mind, presented as a truth table
> ⬇️ `average` / `T` ➡️ 	`None` 	`0` (int) 	`>=1` (int or float)
> `None` 	avg. with `T=2**J` 	no avg. 	avg. with user-set `T`
> `True` 	avg. with `T=2**J` (`DeprecationWarning`) 	`ValueError` 	avg. with user-set `T` (`DeprecationWarning`)
> `False` 	no avg. (`DeprecationWarning`) 	`ValueError` 	`ValueError`
> 
>     * The new default will be `average=None`, `T=None`, and will be backwards compatible with the old default, without warning nor error.
> 
>     * Passing `T=0` and letting `average=None` will be the new way to specify `average=False`.
> 
>     * Already-in-place settings `average=False` and `T>=1` will keep working as intended. The former will raise a `DeprecationWarning`, instructing the user to replace `average=False` by `T=0` before v0.4.
> 
>     * Passing `average=True` together with `T>=1` or `T=None` will raise a `DeprecationWarning`, instructing the user to remove the `average` argument befor v0.4.
> 
>     * Passing `T=0` together with setting `average=False` will raise a `ValueError`, because this is redundant. The `ValueError` will instruct the user to remove the `average` argument immediately.
> 
>     * If both parameters are set (to non-None values), neither of the two parameters will override the other. `T=0, average=True` will error, and same for `T>=1, average=False`.
> 
> 
> We have four possible combinations as of v0.2. With this PR, we will have nine, of which three are errors and three are deprecated. In the future (v0.4), we will remove `average` and only have (9-3-3) = 3 branches: `T=None`, `T=0`, and `T>=1`.
> 
> I haven't included this in the truth table, but cases `T<0` and `0<T<1` should also lead to `ValueError`'s.
> 
> This is a good first step towards supporting `T="global"` in v0.4

excellent - i've resolved these